BookNest/app.js

```diff
@@ -0,0 +1,388 @@
+//includes 
+const express = require('express')
+
+//init app & set port
+const app = express();
+const session = require('express-session');
+const router = express.Router();
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+const multer = require('multer');
+
+const flash = require('connect-flash');
+
+
+// Configure session middleware
+app.use(session({
+    secret: 'your-secret-key', // Change this to a strong secret
+    resave: false,
+    saveUninitialized: true,
+}));
+
+// Initialize flash messages
+app.use(flash());
+
+// Make flash messages accessible in templates
+app.use((req, res, next) => {
+    res.locals.flashMessage = req.flash('info'); // or 'error', depending on your usage
+    next();
+});
+const port = 3000;
+
+app.set('view engine', 'ejs');
+
+//load frontend css & js
+app.use(express.static(__dirname + "/public"));
+
+const path = require('path');
+
+// Middleware to parse form data
+app.use(express.urlencoded({ extended: true }));
+app.use(express.static('public')); // Serve static files
+
+// Middleware to serve static files
+app.use('/uploads', express.static(path.join(__dirname, 'publics', 'uploads')));
+
+// Initialize the cart
+
+// Initialize the cart on first visit
+app.use((req, res, next) => {
+    if (!req.session.cart) {
+        req.session.cart = [];
+    }
+    next();
+});
+
+const sqlite3 = require('sqlite3').verbose();
+const db = new sqlite3.Database('Backend/Database.db');
+
+// Create books table if it doesn't exist
+
+db.serialize(() => {
+    // db.run('INSERT INTO users (email, password) VALUES (?, ?)', [email, password], (err) => {
+    //   if (err) {
+    //     console.error('Database error:', err.message);
+    //   }
+    // });
+ 
+    db.run(`
+        CREATE TABLE IF NOT EXISTS users (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  username TEXT UNIQUE,
+  email TEXT,
+  password TEXT
+);
+      `);
+      });    
+db.run(`
+  CREATE TABLE IF NOT EXISTS books (
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    name TEXT NOT NULL,
+    author TEXT NOT NULL,
+    price REAL,
+    coverImage TEXT
+  )
+`);
+
+db.run(`
+    CREATE TABLE IF NOT EXISTS orders (
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    customer_name TEXT NOT NULL,
+    address TEXT NOT NULL,
+    city TEXT NOT NULL,
+    zip TEXT NOT NULL,
+    total_price REAL NOT NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    status TEXT DEFAULT 'Pending'
+);
+
+CREATE TABLE IF NOT EXISTS order_items (
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    order_id INTEGER NOT NULL,
+    book_id INTEGER NOT NULL,
+    book_name TEXT NOT NULL,
+    author TEXT NOT NULL,
+    price REAL NOT NULL,
+    coverImage TEXT,
+    FOREIGN KEY (order_id) REFERENCES orders (id) ON DELETE CASCADE
```

Comment:
might be best not to use both db browser and js to create tables to avoid potential conflicts